app/model/block.py

```diff
@@ -3,7 +3,20 @@ def __init__(self):
         self.id = None
         self.title = None
         self.sections = []
+        self.container = None
 
     def add_section(self, section):
         if section not in self.sections:
             self.sections.append(section)
+            section.container = self
+
+    def get_item_by_id(self, id):
```

Comment:
Agreed on all of that.  It makes sense to move this functionality into a single location that can be shared between all the model instances, and I thought as much when I wrote it.  Not sure yet whether to move it into a BaseModel class, have it as a Mixin, or move it out into another class yet.  There are more of these types of methods that will be added to model classes, so we're currently mulling over the best approach.